### ISSUE TITLE
Can't log in with these services - GDrive, OneDrive, Box

### ISSUE THREAD

**Issue by eivanhoe** [clementine-player/Clementine#5646]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [x ] I checked the issue tracker for similar issues
- [x] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Arch Linux
- Clementine version: 1.3.1

### Steps to reproduce the problem (only for bugs)

I'm trying to access the cloud services such as GDrive, OneDrive and Box.

I was able to authorise the services to allow clementine to access the music files. After allowing it, the window in Clementine just stays "Login". I was able to login before.

**Comment by tamasd** [comment]
Happens on Kubuntu as well. It opens up the browser, but the OAuth url does not contain the port, so it tries to use localhost directly, which doesn't work. If I put in the port number, the browser window says that it works, but the settings panel doesn't change, so I can't save the credentials.

**Comment by 49e94b8f256530dc0d41f740dfe8a4c1** [comment]
I just troubleshooted this 
```
[user@fedora-desktop ~]$ netstat -tnlp
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:63342         0.0.0.0:*               LISTEN      5835/java           
tcp        0      0 127.0.0.1:39663         0.0.0.0:*               LISTEN      17946/clementine    
tcp        0      0 127.0.0.1:35419         0.0.0.0:*               LISTEN      17946/clementine    
tcp        0      0 127.0.0.1:6942          0.0.0.0:*               LISTEN      5835/java  
```

I got redirected to localhost:80:

>http://localhost/?code=abc123

adding the port worked though:

>http://localhost:39663/?code=abc123

the fix should be easy I think, just add the port wherever the REDIRECT_URL is being built

Very strange that this issue is buried down here.

**Comment by timfel** [comment]
Adding the port manually (no longer?) works, because the service complains the URL doesn't match anymore:
```
15:32:31.654 ERROR OAuthenticator:136               Failed to get access token "{\"error\":\"invalid_grant\",\"error_description\":\"The provided value for the 'redirect_uri' is not valid. The value must exactly match the redirect URI used to obtain the authorization code.\",\"correlation_id\":\"6d85d6df-6784-4132-8b59-cc8f4cd939ba\"}"
```

**Comment by szul** [comment]
Assuming I'm having the same issue. After logging in via OneDrive, I'm redirected to a localhost without a port number and get an error page.

**Comment by Malchior95** [comment]
Adding the port manually does not work on Ubuntu 20.04. I get the Information: 
Success!
Please close your browser and return to Clementine.
When I return to Clementine, I am still not logged in.

**Comment by timfel** [comment]
> Adding the port manually does not work on Ubuntu 20.04. I get the Information:
> Success!
> Please close your browser and return to Clementine.
> When I return to Clementine, I am still not logged in.

Yes, the verbose output of Clementine tells you that it doesn't work, because the redirect URL has to match precisely the one that's transmitted to the API service. See my comment above

**Comment by andyrushtonprogramming** [comment]
I bit of a follow-up on this. I get the same problem: I'm on Linux - specifically Ubuntu 20.04 LTS - Clementine Version 1.4 rc1, the redirect fails as described here.
I tried on Windows, official release from the clementine website v1.3.1, that allowed me to login to Onedrive, the redirect worked, but that version could not find the music (all mp3s in a folder called "Music"). I tried the lastest build 1.4.0rc1-373-gba8fc09a6 on Windows and the redirect does not work on that.
So it looks like this feature hasn't worked for a while, for two different reasons.

**Comment by Yahx3** [comment]
Also getting an issue for box. Won't let me login or anything:

> Error: redirect_uri_missing
> >Show Error Details:
>     response_type=code
>     redirect_uri=https://clementine-data.appspot.com/skydriveclient_id=gbswb9wp7gjyldc3qrw68h2rk68jaf4h
>     client_id=gbswb9wp7gjyldc3qrw68h2rk68jaf4h

**Comment by timfel** [comment]
FYI, this works again for me on Ubuntu with version 1.4.0rc1-568-gc51d2f954

**Comment by sirber** [comment]
I'm on 1.4.0rc1 and it does't work, on Linux Mint.

**Comment by sirber** [comment]
Version 1.3.1 from flatpak works!

**Comment by Mateus109** [comment]
This issue still persists, 2 years later.
